packages/react-native/ReactAndroid/src/main/jni/react/turbomodule/ReactCommon/TurboModuleManager.cpp

```diff
@@ -199,7 +199,7 @@ std::shared_ptr<TurboModule> TurboModuleManager::getTurboModule(
                   "getBindingsInstaller");
       auto installer = getBindingsInstaller(moduleInstance);
       if (installer) {
-        installer->cthis()->installBindings(runtime);
+        installer->cthis()->installBindings(runtime, jsCallInvoker_);
```

Comment:
we can actually get the callInvoker off the ReactContext, but i can see how that's inapplicable here... but i'm reading all of this and i'm starting to feel like a C++ TM is more appropriate here? what's the current limitation?

Reply:
I need it in [`NitroModules.kt (init)`](https://github.com/mrousavy/nitro/blob/2bf4e6110738f64bf9eb5d47da3b200e531e1cd8/packages/react-native-nitro-modules/android/src/main/java/com/margelo/nitro/NitroModules.kt#L21) - this will (optionally) be passed along to other Nitro Modules. But it goes through C++ first.

Reply:
> I need it in [`NitroModules.kt (init)`](https://github.com/mrousavy/nitro/blob/2bf4e6110738f64bf9eb5d47da3b200e531e1cd8/packages/react-native-nitro-modules/android/src/main/java/com/margelo/nitro/NitroModules.kt#L21) - this will (optionally) be passed along to other Nitro Modules. But it goes through C++ first.

thanks! is the idea that we want to replace install with installJSIBindings?

if so, is it possible to keep the current native pipeline to pass down the CallInvoker (i.e. do what you have now), but use installJSIBindings for the runtime installation? or will there be some sort of timing issue?

i can see how ergonomically that might feel weird, what's your feeling on that?

Reply:
Yep - I want to have a clear, public and guaranteed to be future-proof API to install Nitro into the jsi::Runtime.
The current approach _works_, but having to annotate with `@OptIn(FrameworkAPI::class)` doesn't sound very public-API-like to me?

Reply:
> if so, is it possible to keep the current native pipeline to pass down the CallInvoker (i.e. do what you have now), but use installJSIBindings for the runtime installation? or will there be some sort of timing issue?

Well `installJSIBindings` (or `BindingsInstaller` on Android) is accessed _before_ I can call `install()` on the module. So I don't have a CallInvoker yet at that point on Android.